usr/share/rear/lib/config-functions.sh

```diff
@@ -16,18 +26,63 @@ function SetOSVendorAndVersion () {
     # The test must match OS_VENDOR=generic or OS_VERSION=none in default.conf:
     if test "$OS_VENDOR" = generic -o "$OS_VERSION" = none ; then
 
-        # Recent Linux distro's with systemd has the /etc/os-release file
+        # Recent Linux distros with systemd has the /etc/os-release file
         # Try to find all the required information from that file
         if [[ -f /etc/os-release ]] ; then
-            grep -q -i 'fedora' /etc/os-release && OS_VENDOR=Fedora
-            grep -q -i -E '(centos|redhat|scientific|oracle)' /etc/os-release && OS_VENDOR=RedHatEnterpriseServer
-            grep -q -i 'suse' /etc/os-release && OS_VENDOR=SUSE_LINUX
-            grep -q -i 'debian' /etc/os-release && OS_VENDOR=Debian
-            grep -q -i -E '(ubuntu|linuxmint)' /etc/os-release && OS_VENDOR=Ubuntu
-            grep -q -i 'arch' /etc/os-release && OS_VENDOR=Arch
+            local ID ID_LIKE
+            ID="$(get_var_from_file /etc/os-release ID)"
+            ID_LIKE="$(get_var_from_file /etc/os-release ID_LIKE)"
+
+            # Deal with derivatives first:
+            case "$ID_LIKE" in
+                (*arch*)
+                    OS_VENDOR=Arch_Linux
+                    ;;
+                (*centos*|*rhel*)
+                    OS_VENDOR=RedHatEnterpriseServer
+                    ;;
+                (*debian*)
+                    OS_VENDOR=Debian
+                    ;;
+                # Fedora must be after RHEL to distinguish RHEL derivatives.
+                (*fedora*)
+                    OS_VENDOR=Fedora
+                    ;;
+                (*suse*)
+                    OS_VENDOR=SUSE_LINUX
+                    ;;
+                (*ubuntu*)
+                    OS_VENDOR=Ubuntu
+                    ;;
+            esac
+
+            # And then use the actual ID.  This may reset OS_VENDOR if it was
+            # already set. E.g. Ubuntu is like Debian but ReaR treats it as
+            # a separate OS_VENDOR.
+            case "$ID" in
+                (arch)
+                    OS_VENDOR=Arch_Linux
+                    ;;
+                (debian)
+                    OS_VENDOR=Debian
+                    ;;
+                (fedora)
+                    OS_VENDOR=Fedora
+                    ;;
+                (rhel|ol)
+                    # Oracle Linux specifies only Fedora and not RHEL in ID_LIKE=
+                    OS_VENDOR=RedHatEnterpriseServer
+                    ;;
+                (suse)
+                    OS_VENDOR=SUSE_LINUX
+                    ;;
+                (ubuntu)
+                    OS_VENDOR=Ubuntu
+                    ;;
+            esac
 
             local VERSION_ID
-            eval "$(grep "^VERSION_ID=" /etc/os-release)"
+            VERSION_ID="$(get_var_from_file /etc/os-release VERSION_ID)"
             contains_visible_char "$VERSION_ID" && OS_VERSION="$VERSION_ID"
```

Comment:
With the get_var_from_file function
you can even use lowercase local variables like
```
local version_ID
version_ID="$( get_var_from_file /etc/os-release version_ID )"
contains_visible_char "$version_ID" && OS_VERSION="$version_ID"
```
Feel free to understand "lowercase" with a grain of salt
i.e. use `version_id` (strictly lowercase)
or `version_ID` (lowercase "cum grano salis")
as you like.

Reply:
Addressed in 55dd4e5 and 3aa6f15.